src/EditorFeatures/Core/ExternalAccess/VSTypeScript/VSTypeScriptProjectContextHandler.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.DocumentChanges;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript;
+
+[ExportStatelessLspService(typeof(GetTextDocumentWithContextHandler), ProtocolConstants.TypeScriptLanguageContract), Shared]
+internal class VSTypeScriptProjectContextHandler : GetTextDocumentWithContextHandler
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public VSTypeScriptProjectContextHandler()
```

Comment:
We don't actually need an implementation here?

Reply:
The base implementation works for TS, so we don't need any custom behavior here.

Reply:
@dibarbet Does any language ever need to customize it? I'm not sure why this isn't just an automatic for everything.

Reply:
Any roslyn based one shouldn't need to.  But some handlers are different so the MEF exports are per server are needed.  I did try and add support for 'export this for all server' but it ended up being more complicated than it was worth when having to deal with restarts, etc.